websocket.go

```diff
@@ -222,8 +218,12 @@ func (w *WebSocket) stop() {
 	if !w.started {
 		return
 	}
-	log.Lvl3("Stopping", w.server.Server.Addr)
-	w.server.Stop(100 * time.Millisecond)
+	log.Lvl3("Stopping", w.server.Addr)
+
+	d := time.Now().Add(100 * time.Millisecond)
+	ctx, _ := context.WithDeadline(context.Background(), d)
+	w.server.Shutdown(ctx)
+
 	<-w.startstop
```

Comment:
Defer will trigger when the function returns, but the shutdown is not expected to be done by then, it would shortcut the timeout. Still reading and thinking to see if I can convince myself either that the timeout is not useful or that the cancel can be called so fast.

Reply:
`w.server.Shutdown(ctx)` is blocking so it won't defer until the timeout or the graceful shutdown. Usually, if you have a context, that means the function will block and the context gives you a chance to abort the operation if necessary, like using a timeout in that case. But the cancel function should *always* be called at least once.

Reply:
You can forget about the `SetKeepAlivesEnabled` however as you're shutting down anyway.

Reply:
I figured out my confusion, I was looking at the signalling via w.startstop, and got confused thinking that WebSocket.Stop() was trying to be asynchronous, when in fact it blocking and sa @Gilthoniel says the call to net/http.Shutdown is blocking (until the context expires) as well.